coremltools/test/neural_network/test_graph_passes.py

```diff
@@ -203,6 +203,77 @@ def test_conv_crop_bn_relu_to_conv_bn_relu_crop(self):
         np.testing.assert_equal('activation', spec.layers[2].WhichOneof('layer'))
         np.testing.assert_equal('crop', spec.layers[3].WhichOneof('layer'))
 
+    def test_redundant_transposes(self):
+        input_features = [('data', datatypes.Array(1, 10, 10))]
+        output_features = [('out', None)]
+        builder = neural_network.NeuralNetworkBuilder(input_features, output_features)
+        # These transposes together are the identity.
+        builder.add_transpose(name='t1',
+                              axes=[2, 0, 1],
+                              input_name='data',
+                              output_name='t1_out')
+        builder.add_transpose(name='t2',
+                              axes=[1, 2, 0],
+                              input_name='t1_out',
+                              output_name='out')
+        # Transpose1 -> Transpose2
+        spec = builder.spec.neuralNetwork
+        np.testing.assert_equal('transpose', spec.layers[0].WhichOneof('layer'))
+        np.testing.assert_equal('transpose', spec.layers[1].WhichOneof('layer'))
+        # Run the removal pass.
+        remove_redundant_transposes(builder.spec)
+        # Redundant transposes removed.
+        np.testing.assert_equal(len(spec.layers), 0)
+
+        builder = neural_network.NeuralNetworkBuilder(input_features, output_features)
+        # These transposes are not inverses.
+        builder.add_transpose(name='t1',
+                              axes=[2, 0, 1],
+                              input_name='data',
+                              output_name='t1_out')
+        builder.add_transpose(name='t2',
+                              axes=[2, 0, 1],
+                              input_name='t1_out',
+                              output_name='out')
+        # Transpose1 -> Transpose2
+        spec = builder.spec.neuralNetwork
+        np.testing.assert_equal('transpose', spec.layers[0].WhichOneof('layer'))
+        np.testing.assert_equal('transpose', spec.layers[1].WhichOneof('layer'))
+        # Run the removal pass.
+        remove_redundant_transposes(builder.spec)
+        # Transpose1 -> Transpose2
+        np.testing.assert_equal('transpose', spec.layers[0].WhichOneof('layer'))
+        np.testing.assert_equal('transpose', spec.layers[1].WhichOneof('layer'))
+
+        input_features = [('data', datatypes.Array(1, 1, 10, 10, 3))]
```

Comment:
could we share code to combine the testing, something like loop over a list of `[[input_shape, transpose_1, transpose_2]]`, e.g.:` [[(1,10,10), (2,0,1), (2,0,1)], [(1,1,10,10,3),(2,4,1,0,3),(3,2,0,4,1)],[...]]` ? 
(similar to the test [here](https://github.com/apple/coremltools/blob/963d9d3c166403de55ae91002d500c9f806d7300/coremltools/test/neural_network/test_numpy_nn_layers.py#L708), other than that test is random, but here there can be deterministic test cases)